aeon/io/reader.py

```diff
@@ -185,6 +186,22 @@ class Encoder(Harp):
     def __init__(self, pattern):
         super().__init__(pattern, columns=["angle", "intensity"])
 
+    def read(self, file, downsample=True):
+        """Reads encoder data from the specified Harp binary file, and optionally downsamples
+        the frequency to 50Hz.
+        """
+        data = super().read(file)
+        if downsample is True:
+            # resample requires a DatetimeIndex so we convert early
+            data.index = aeon_time(data.index)
+
+            first_index = data.first_valid_index()
+            if first_index is not None:
+                # since data is absolute angular position we decimate by taking first of each bin
+                chunk_origin = chunk(first_index)
+                data = data.resample('20ms', origin=chunk_origin).first()
+        return data
```

Comment:
I'm not sure I agree with repeatedly wrapping and rethrowing exceptions as this will hide the true call stack. Until we know exactly what errors will be thrown I would err on the side of not trying to second guess the code.

Regarding validation of the `downsample` parameter, checking only for `downsample is True` was deliberate. In the future we could pass non-boolean values into `downsample`, for example indicating the specific resampling frequency. For now we want to accept `None` and `False` as possible alternatives. It is true maybe we could be more thorough in the validation, although we don't expect this to be a parameter which should be much used in practice.

@ttngu207 @jkbhagatio any thoughts?

Reply:
> I'm not sure I agree with repeatedly wrapping and rethrowing exceptions as this will hide the true call stack. Until we know exactly what errors will be thrown I would err on the side of not trying to second guess the code.
> 
> Regarding validation of the `downsample` parameter, checking only for `downsample is True` was deliberate. In the future we could pass non-boolean values into `downsample`, for example indicating the specific resampling frequency. For now we want to accept `None` and `False` as possible alternatives. It is true maybe we could be more thorough in the validation, although we don't expect this to be a parameter which should be much used in practice.
> 
> @ttngu207 @jkbhagatio any thoughts?

Agreed, if it does error out, let it error out and throw the full error stack.
Unless we are expecting and explicitly catching a specific type of error, I'd agree not to keep wrapping code in generic `try, except Exception` block